### ISSUE TITLE
Missing optimization flags

### ISSUE THREAD

**Issue by pubby** [andrewmcwattersandco/programming-language-benchmarks#1]
Looks like you're not compiling the C and C++ files with the -O2 flag, meaning the compiler isn't optimizing anything. Dunno about other languages.

**Comment by andrewmcwatters** [comment]
We do not use specific flags for the benchmarks. If a language tutorial from a primary source, for example, specifies it as a common practice, we might be inclined to specify it.

From _README.md_:
> Use idiomatic code/Do not optimize

**Comment by pubby** [comment]
It is idiomatic to compile with optimizations on.

**Comment by andrewmcwatters** [comment]
OK, we'll consider doing it then. Is there a documentation page or article from gcc or clang that suggests a recommended optimization level?

The reason I ask is I'm hoping to make sure the benchmarks look like what most developers would do in that given language, so that the benchmark results don't end up being contrived.

I think I'm wrong about this in the first place on Windows, because I think new projects in Visual Studio code by default have a particular optimization level set.

**Comment by andrewmcwatters** [comment]
Optimization best practices (https://docs.microsoft.com/en-us/cpp/build/optimization-best-practices?view=msvc-170) recommends PGO first, and then **/O** compiler options. It seems like new projects by default do not turn these options on, actually. So maybe I'm mistaken. I imagine there are some similar articles for gcc and clang.

Either way, thank you for taking the time to file an issue about this. I appreciate it, and will spend some more time reading about this.

I'm aiming for what, say, 80%, or a majority of of developers would be doing. So if I find that most people set particular optimization levels, then we'll do that. 👍

**Comment by pubby** [comment]
For Clang and GCC just use -O2, as that's what 90% of developers are using. For MSVC use /O2. Compiling with no flags is only intended for debug builds, not release builds.

I reckon PGO is too niche, so I would ignore it.

The relevant documentation is:

https://gcc.gnu.org/onlinedocs/gcc/Optimize-Options.html#Optimize-Options

https://docs.microsoft.com/en-us/cpp/build/reference/o-options-optimize-code?view=msvc-170

**Comment by andrewmcwatters** [comment]
I agree about PGO, I hadn't heard of it until now. I'll do more reading. If -O2 is the go-to optimization level, we'll use that.

**Comment by andrewmcwatters** [comment]
Gentoo recommends -O2. https://wiki.gentoo.org/wiki/GCC_optimization#-O I'll see if I can find more references.
